javascript/javascript3/week1/weatherApp/weather.js

```diff
@@ -0,0 +1,144 @@
+    todayDate = new Date();
+    todaysDate.innerHTML = todayDate.toLocaleDateString(undefined, options);
+    currentTime.innerHTML = todayDate.toLocaleTimeString('en-US');
+}
+setInterval(getcurrentDetails, 1000); //to update time every second
+
+
+//get user entered city location and set the cityname in the API link when get the city location weather button clicked
+const weatherButton = getHtmlMarkup('weatherButtonMarkup');
+
+weatherButton.addEventListener('click', () => {
+    const cityInput = getHtmlMarkup('cityInputMarkup').value.trim();
+    (cityInput)? getWeatherByCity(cityInput) : alert('Please Enter The City Name');;
+});
+
+function getWeatherByCity(city) {
+    weatherApi = `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=metric`;
+    displayWeatherData(weatherApi);
+}
+
+
+//get the current location coordinates and set the coordinates in the API link when get the current location weather button clicked
+const currentlocationweather = getHtmlMarkup('currentlocationweatherButton');
+let locationLatitude;
+let locationLongitude;
+
+currentlocationweather.addEventListener('click', function() {
+    navigator.geolocation.getCurrentPosition(success,error);
+});
+
+function success(pos) {
+    var crd = pos.coords;
+    locationLatitude = crd.latitude;
+    locationLongitude = crd.longitude;
+    
+    weatherApi = `https://api.openweathermap.org/data/2.5/weather?lat=${locationLatitude}&lon=${locationLongitude}&appid=${apiKey}&units=metric`;
+    displayWeatherData(weatherApi);
+}
+
+function error(err) {
+    console.warn(`ERROR(${err.code}): ${err.message}`);
+}
+
+
+//Displaying all the weather details in the html page
+function displayWeatherData(weatherApi) {
+    fetch(weatherApi)
+    .then(response => response.json())
+    .then(weatherData => {
+        
+        //Storing the cityname in the local storage
+        localStorage.setItem('cityName', weatherData.name);
+        
+        //City Name
+        getHtmlMarkup('cityname').innerHTML = `${weatherData.name}`;
+       
+        //Temperature in celsius
+        getHtmlMarkup('temperature').innerHTML = `${weatherData.main.temp} °C`;
+  
+        //Weathericon
+        getHtmlMarkup('weatherIcon').src = `http://openweathermap.org/img/wn/${weatherData.weather[0].icon}@2x.png`;
+        
+       //Cloud Description and percentage
+        getHtmlMarkup('cloudyDetails').innerHTML = `${weatherData.weather[0].description.toUpperCase()} - ${weatherData.clouds.all}%`;
+       
+        //feels like temperature in celsius
+        getHtmlMarkup('feelsLike').innerHTML = `Feels_Like: ${weatherData.main.feels_like} °C`;
+      
+        //Wind speed in meter/second
+        getHtmlMarkup('windSpeed').innerHTML = `Wind Speed: ${weatherData.wind.speed} m/sec`;
+        
+        
+        getHtmlMarkup('timeinfo').innerHTML = `Below time is the time in city location ${weatherData.name.toUpperCase()} and not displayed according to your current location`
```

Comment:
from line 85 to 103 
```suggestion
      const timeInfo =  getHtmlMarkup('timeinfo');
       timeInfo.innerHTML = `Below time is the time in city location ${weatherData.name.toUpperCase()} and not displayed according to your current location`
```
This is more understandable and easy to read